eclair-core/src/main/scala/fr/acinq/eclair/payment/relay/PostRestartHtlcCleaner.scala

```diff
@@ -70,7 +70,12 @@ class PostRestartHtlcCleaner(nodeParams: NodeParams, register: ActorRef, initial
     val nonStandardRelayedOutHtlcs: Map[Origin, Set[(ByteVector32, Long)]] = nodeParams.pluginParams.collect { case p: CustomCommitmentsPlugin => p.getHtlcsRelayedOut(htlcsIn, nodeParams, log) }.flatten.toMap
     val relayedOut: Map[Origin, Set[(ByteVector32, Long)]] = getHtlcsRelayedOut(channels, htlcsIn) ++ nonStandardRelayedOutHtlcs
 
-    val notRelayed = htlcsIn.filterNot(htlcIn => relayedOut.keys.exists(origin => matchesOrigin(htlcIn.add, origin)))
+    val notRelayed = htlcsIn.filterNot(htlcIn => {
+      // If an HTLC has been relayed and then settled downstream, it will not have a matching entry in relayedOut.
+      // When that happens, there will be an HTLC settlement command in the pendingRelay DB, and we will let the channel
+      // replay it instead of sending a conflicting command.
+      relayedOut.keys.exists(origin => matchesOrigin(htlcIn.add, origin)) || nodeParams.db.pendingCommands.listSettlementCommands(htlcIn.add.channelId).exists(_.id == htlcIn.add.id)
+    })
```

Comment:
This would avoid too many calls to the database (not that we are expecting a lot but still).

```suggestion
    val settledHtlcs: Set[(ByteVector32, Long)] = nodeParams.db.pendingCommands.listSettlementCommands().map { case (channelId, cmd) => (channelId, cmd.id) }.toSet

    val notRelayed = htlcsIn.filterNot { htlcIn =>
      // If an HTLC has been relayed and then settled downstream, it will not have a matching entry in relayedOut.
      // When that happens, there will be an HTLC settlement command in the pendingRelay DB, and we will let the channel
      // replay it instead of sending a conflicting command.
      relayedOut.keys.exists(origin => matchesOrigin(htlcIn.add, origin)) || settledHtlcs.contains(htlcIn.add.channelId, htlcIn.add.id)
    }
```

Reply:
Done in https://github.com/ACINQ/eclair/pull/1817/commits/ee9a1e03f284e3e2250b086cffe369d3d5751017